client/asset/dcr/dcr.go

```diff
@@ -4527,3 +4546,178 @@ func float64PtrStr(v *float64) string {
 	}
 	return strconv.FormatFloat(*v, 'f', 8, 64)
 }
+
+func (dcr *ExchangeWallet) ConfirmRedemption(coinID dex.Bytes, redemption *asset.Redemption) (*asset.ConfirmRedemptionStatus, error) {
+	txHash, _, err := decodeCoinID(coinID)
+	if err != nil {
+		return nil, err
+	}
+
+	var secretHash [32]byte
+	copy(secretHash[:], redemption.Spends.SecretHash)
+	dcr.mempoolRedeemsMtx.RLock()
+	mRedeem, have := dcr.mempoolRedeems[secretHash]
+	dcr.mempoolRedeemsMtx.RUnlock()
+
+	var deleteMempoolRedeem bool
+	defer func() {
+		if deleteMempoolRedeem {
+			dcr.mempoolRedeemsMtx.Lock()
+			delete(dcr.mempoolRedeems, secretHash)
+			dcr.mempoolRedeemsMtx.Unlock()
+		}
+	}()
+
+	var abandonTx bool
+	tx, err := dcr.wallet.GetTransaction(dcr.ctx, txHash)
+	if err == nil {
+		if have && mRedeem.txHash == *txHash {
+			if tx.Confirmations == 0 && time.Now().After(mRedeem.firstSeen.Add(maxRedeemMempoolAge)) {
+				// Transaction has been sitting in the mempool
+				// for a long time now. Attempt to abandon.
+				abandonTx = true
+				if err := dcr.wallet.AbandonTransaction(dcr.ctx, txHash); err != nil {
+					dcr.log.Warnf("Unable to abandon redemption mempool transaction %v: %w.", txHash, err)
+				} else {
+					dcr.log.Infof("Abandoned redemption mempool transaction: %v.", txHash)
+					deleteMempoolRedeem = true
+				}
+			}
+		} else {
+			// Will hit this if node was restarted with an actively
+			// redeeming swap.
+			dcr.mempoolRedeemsMtx.Lock()
+			dcr.mempoolRedeems[secretHash] = &mempoolRedeem{txHash: *txHash, firstSeen: time.Now()}
+			dcr.mempoolRedeemsMtx.Unlock()
+		}
+		if !abandonTx {
+			if tx.Confirmations >= reqiredRedeemConfirms {
+				deleteMempoolRedeem = true
+			}
+			return &asset.ConfirmRedemptionStatus{
+				Confs:  uint64(tx.Confirmations),
+				Req:    reqiredRedeemConfirms,
+				CoinID: coinID,
+			}, nil
+		}
+	}
+
+	if !abandonTx && !errors.Is(err, asset.CoinNotFoundError) {
+		return nil, fmt.Errorf("problem searching for redemption transaction %s: %w", txHash, err)
+	}
+
+	// Redemption transaction is missing from the point of view of our node!
+	// Unlikely, but possible it was redeemed by another transaction. We
+	// assume a contract past its locktime cannot make it here, so it must
+	// not be refunded. Check if the contract is still an unspent output.
+
+	txHash, vout, err := decodeCoinID(redemption.Spends.Coin.ID())
```

Comment:
`txHash` is used above for the redemption tx. How about `swapHash` or something here to differentiate? Same in **btc**.

Reply:
Changed to `swapHash` in both places.